File: code/DISTMAT.m
%X: m*n
%Dis:  m*m

function Dis = DISMAT(X)
[m,n] = size(X);
A = X*X';
l = diag(A);
Dis = sqrt(repmat(l,[1,m])+ repmat(l',[m,1]) - 2*A);
							  
end

File: code/DISTMAT.R
DISTMAT <- function (X){
  n <- ncol(X)
  m <- nrow(X)
  X <- as.matrix(X)
  A <- X %*% t(X)
  l <- diag(A)
  Dis <- sqrt(matrix(l, nrow = m, ncol =m)+ t(matrix(l, nrow = m, ncol =m)) - 2*A)
  return(Dis)
}
